.grit/patterns/es6_imports.md

```diff
@@ -9,18 +9,29 @@ Converts `require` statements to ES6-style `import`.
 tags: #js, #es6, #migration, #cjs, #commonjs
 
 ```grit
+engine marzano(0.1)
+language js
+
 or {
     `const $sentry = require('@sentry/node')` => `import * as $sentry from '@sentry/node'`,
-    `const { $imports } = require($source)` => `import { $transformed } from "$source"` where {
-        $imports <: some bubble($transformed) {ObjectProperty(key=$key, value=$val) where {
-            $transformed = [...$transformed, `$key as $val`]
-        } }
-    },
-    `const $import = require($source).default` => `import $import from "$source"`,
-    `const $import = require($source).$foo` => `import { $foo as $import } from "$source"`,
-    `const $import = require($source)` => `import $import from "$source"`, // this relies on healing for correctness
     // see https://github.com/motdotla/dotenv#how-do-i-use-dotenv-with-import
-    `require("dotenv").config($config)` => [`import * as dotenv from 'dotenv'`, `dotenv.config($config)`]
+    `require("dotenv").config($config)` => `import * as dotenv from 'dotenv';\ndotenv.config($config)`,
+
+    `const { $imports } = require($source)` where {
+        $import_list = [],
+        $imports <: some bubble($import_list) {
+            or {
+                shorthand_property_identifier_pattern() as $key where $import_list += $key,
+                pair_pattern($key, $value) where $import_list += `$key as $value`
+            }
+        },
+        $transformed = join(list = $import_list, separator = ", "),
+    } => `import { $transformed } from $source`,
+    `const $import = require($source).default` => `import $import from $source`,
+    `const $name = require($source).$from` => `import { $name } from $source` where { $name <: $from},
```

Comment:
Why doesn't

```
`const $name = require($source).$from` => `import { $from } from $source`
```
work?

Reply:
It does. The point of this is to ensure $name and $from are identical.